nixos/modules/services/web-apps/stash.nix

```diff
@@ -0,0 +1,116 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.stash;
+in
+{
+  options = {
+    services.stash = {
+      enable = mkEnableOption "an organizer for your porn, written in Go";
+
+      package = mkPackageOption pkgs "stash" { };
+
+      user = mkOption {
+        type = types.str;
+        default = "stash";
+        description = "User account under which stash runs.";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "stash";
+        description = "Group under which stash runs.";
+      };
+
+      ip = mkOption {
+        type = types.str;
+        default = "0.0.0.0";
+        example = "::1";
+        description = "The ip that the server should bind to.";
+      };
+
+      port = mkOption {
+        type = types.port;
+        default = 9999;
+        description = "The port that the server should listen on.";
+      };
+
+      openFirewall = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Open ports in the firewall for the stash web interface.";
+      };
+
+      readWritePaths = mkOption {
+        type = types.listOf types.path;
+        default = [ ];
+        description = "List of read and write paths.";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.stash = {
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network.target" ];
+      environment = {
+        STASH_HOST = cfg.ip;
+        STASH_PORT = toString cfg.port;
+        STASH_GENERATED = "%S/stash/generated";
+        STASH_METADATA = "%S/stash/metadata";
+        STASH_CACHE = "%S/stash/cache";
+      };
+      preStart = ''mkdir -p -- "$STASH_GENERATED" "$STASH_METADATA" "$STASH_CACHE"'';
```

Comment:
should this not be systemd tmpfile rules

Reply:
I'd love to see an attempt before dismissing it or some more confidence that its not a viable option as this use if `mkdir` feels hacky.